dessia_common/document_generator.py

```diff
@@ -1,44 +1,103 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """ Write document file. """
-from typing import List, Tuple
+from typing import List
 
 import docx
 
+from dessia_common.files import BinaryFile
+
+
+class HeaderFooter:
+    """ Represents a header or footer in a docx document. """
+
+    def __init__(self, text: str, align: str = 'center'):
+        self.text = text
+        self.align = align
+
+    def add_to_section(self, section, is_header):
+        """ Add the header/footer to the document. """
+        if is_header:
+            header = section.header
+        else:
+            footer = section.footer
+        paragraph = header.add_paragraph() if is_header else footer.add_paragraph()
+        paragraph.text = self.text
+        paragraph.alignment = getattr(docx.enum.text.WD_ALIGN_PARAGRAPH, self.align.upper())
+
+
+class Header(HeaderFooter):
+    """ Represents a header in a docx document. """
+
+    def add_to_document(self, document: docx.Document):
+        """ Add the header to the document. """
+        section = document.sections[-1]
```

Comment:
Why is this line not mutualized if it is the same for header and footer ?

Reply:
I think it's much more flexible like that, in case one of the classes changes afterwards.

what do you think ?